core/src/main/java/org/fao/geonet/kernel/security/openidconnect/OIDCConfiguration.java

```diff
@@ -0,0 +1,225 @@
+     * You can specify via setRoleConverterString in the form:
+     * "OIDCServerRole1=GNProfile,OIDCServerRole2=GROUP:PROFILE"
+     */
+    public Map<String, String> roleConverter = new HashMap<>();
+
+    /**
+     * All users who login via the OIDC will have this profile (at a minimum).
+     * This is useful to allow ALL users in an org to login to GN without having to setup all users.
+     * Typically, this should be GUEST or REGISTEREDUSER.  But, you can make it higher (i.e. Editor).
+     */
+    public String minimumProfile = "Guest";
+
+    /**
+     * if true, always update the GN user with information from OIDC.
+     */
+    public boolean userProfileUpdateEnabled = true;
+
+    /**
+     * if true, always update the GN group-profile with information from the OIDC roles.
+     */
+    public boolean userGroupUpdateEnabled = true;
+
+
+    public String clientId;
+    public String clientSecret;
+    public String scopes = null; //null or empty -> all
+    public String roleConverterString = null;
+    public LoginType loginType = LoginType.LINK;
+
+    public String getScopes() {
+        return scopes;
+    }
+
+    public void setScopes(String scopes) {
+        this.scopes = scopes;
+    }
+
+    public List<String> getScopeSet() {
+        if (!StringUtils.hasText(scopes))
+            return null;
+        return Arrays.asList(scopes.split(" "));
+    }
+
+    public String getUserNameAttribute() {
+        return userNameAttribute;
+    }
+
+    public void setUserNameAttribute(String userNameAttribute) {
+        this.userNameAttribute = userNameAttribute;
+    }
+
+    public String getOrganizationProperty() {
+        return organizationProperty;
+    }
+
+    public void setOrganizationProperty(String organizationProperty) {
+        this.organizationProperty = organizationProperty;
+    }
+
+    public String getGroupPermissionSeparator() {
+        return groupPermissionSeparator;
+    }
+
+    public void setGroupPermissionSeparator(String groupPermissionSeparator) {
+        this.groupPermissionSeparator = groupPermissionSeparator;
+    }
+
+    public String getIdTokenRoleLocation() {
+        return idTokenRoleLocation;
+    }
+
+    public void setIdTokenRoleLocation(String idTokenRoleLocation) {
+        this.idTokenRoleLocation = idTokenRoleLocation;
+    }
+
+    public Map<String, String> getRoleConverter() {
+        return roleConverter;
+    }
+
+    public void setRoleConverter(Map<String, String> roleConverter) {
+        this.roleConverter = roleConverter;
+    }
+
+    public Profile getMinimumProfile() {
+        return Profile.findProfileIgnoreCase(minimumProfile);
+    }
+
+    public void setMinimumProfile(String minimumProfile) {
+        this.minimumProfile = minimumProfile;
+    }
+
+    @Override
+    public String getLoginType() {
+        return loginType.toString();
+    }
+
+    public void setLoginType(String loginType) {
+        this.loginType = LoginType.parse(loginType);
```

Comment:
You may want to look at the following

    https://github.com/geonetwork/core-geonetwork/blob/5104515459c85a54e10ff39151ab443e053352d2/core/src/main/java/org/fao/geonet/kernel/security/keycloak/KeycloakConfiguration.java#L125-L141

There are other logintypes.  I'm not sure how loginType.Form would work in openidconnect. I suspect that it should be an invalid option.

Reply:
Yes - I think the only 2 valid options are LINK and AUTOLOGIN.  I put them in the documentation.  I can throw an exception if they aren't one of those...

Reply:
added check